tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,311 @@
+import dataclasses
+import os
+from typing import Optional, cast
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.steps.provision.testcloud
+import tmt.utils
+from tmt.steps.provision.testcloud import GuestTestcloud
+from tmt.utils import field
+from tmt.utils.templates import render_template
+
+DEFAULT_IMAGE_BUILDER = "quay.io/centos-bootc/bootc-image-builder:latest"
+CONTAINER_STORAGE_DIR = tmt.utils.Path("/var/lib/containers/storage")
+
+PODMAN_MACHINE_NAME = 'podman-machine-tmt'
+PODMAN_ENV = tmt.utils.Environment.from_dict({"CONTAINER_CONNECTION": f'{PODMAN_MACHINE_NAME}-root'})
+PODMAN_MACHINE_CPU = os.getenv('TMT_BOOTC_PODMAN_MACHINE_CPU', '2')
+PODMAN_MACHINE_MEM = os.getenv('TMT_BOOTC_PODMAN_MACHINE_MEM', '2048')
+PODMAN_MACHINE_DISK_SIZE = os.getenv('TMT_BOOTC_PODMAN_MACHINE_DISK_SIZE', '50')
```

Comment:
`PODMAN_MACHINE_*` - at first I wanted to convert these into keys in `BootcData`, i.e. make them configurable via fmf and CLI. As they exist here and now it's just environment variable, and that's not consistent with other plugins and their configurables.

But then: it's just `hardware`, we already have ways how a HW requirement should be specified, https://tmt.readthedocs.io/en/stable/spec/hardware.html#cpu, and here we have pretty much the following:

```
hardware:
  cpu:
    processors: ...
  memory: ...
  disk:
    - size: ...
```

It's not really `hardware` key, rather `podman-machine-hwardware`, but the content, the way how HW requirement are given, should be the same. So, I have the following proposal:

* please, drop the environment variables from these: they do not follow the schema we established for plugin keys, https://tmt.readthedocs.io/en/stable/overview.html#plugin-variables, and their very existence suggests it should be possible to modify the number of `podman-machine` CPUs this way - it shouldn't we need to stick to HW specification we already have to remain consistent.
* we're using Pint to help us with units, so we could track the magnitude and unit together, therefore `2048` should be `2048 MB` or something similar: see how `virtual` does it, https://github.com/teemtee/tmt/blob/main/tmt/steps/provision/testcloud.py#L180. The code below would then use `PODMAN_MACHINE_MEM.to('MB').magnitude` when passing the value to commands.
* together, we would end up with something like the following:
  ```python
  DEFAULT_PODMAN_MACHINE_CPU = 2
  DEFAULT_PODMAN_MACHINE_MEM: 'Size' = tmt.hardware.UNITS('2048 MB')
  DEFAULT_PODMAN_MACHINE_DISK_SIZE: 'Size' = tmt.hardware.UNITS('50 GB')
  ```

That would align your code with the rest of tmt, but it would also remove the configurability of these three values, which would be unfortunate, but I strongly believe we should allow it, but not this way. And I would submit a follow-up patch implementing `podman-machine-hardware` key, to open this to users - I don't want to block this PR on it, but I also don't want to let `TMT_BOOTC_PODMAN_MACHINE_MEM` being used, because once it's out there, it's much harder to get rid of it...